File: pkg/dto/multi_cluster_repository.go
package dto

import "github.com/KubeOperator/KubeOperator/pkg/model"

type MultiClusterRepository struct {
	model.MultiClusterRepository
}

type MultiClusterRepositoryCreateRequest struct {
	Name     string `json:"name"`
	Source   string `json:"source"`
	Branch   string `json:"branch"`
	Username string `json:"username"`
	Password string `json:"password"`
}

type MultiClusterRepositoryUpdateRequest struct {
	GitTimeout   int64 `json:"gitTimeout"`
	SyncInterval int64 `json:"syncInterval"`
	SyncEnable   bool  `json:"syncEnable"`
}

type UpdateRelationRequest struct {
	ClusterNames []string `json:"clusterNames"`
	Delete       bool     `json:"delete"`
}

type ClusterRelation struct {
	model.ClusterMultiClusterRepository
	ClusterName string `json:"clusterName"`
}

type MultiClusterSyncClusterLog struct {
	model.MultiClusterSyncClusterLog
	MultiClusterSyncClusterResourceLogs []model.MultiClusterSyncClusterResourceLog `json:"multiClusterSyncClusterResourceLogs"`
	ClusterName                         string                                     `json:"clusterName"`
}

type MultiClusterSyncLogDetail struct {
	model.MultiClusterSyncLog
	MultiClusterSyncClusterLogs []MultiClusterSyncClusterLog `json:"multiClusterSyncClusterLogs"`
}

type MultiClusterSyncLog struct {
	model.MultiClusterSyncLog
}
type MultiClusterRepositoryBatch struct {
	Operation string                   `json:"operation" validate:"required"`
	Items     []MultiClusterRepository `json:"items" validate:"required"`
}

File: ui/src/app/business/multi-cluster/multi-cluster-repository.ts
import {BaseModel, BaseRequest} from "../../shared/class/BaseModel";

export class MultiClusterRepository extends BaseModel {
    name: string;
    source: string;
    message: string;
    status: string;
    branch: string;
    gitTimeout: number;
    syncInterval: number;
    syncEnable: boolean;
}

export class MultiClusterRepositoryCreateRequest {
    name: string;
    source: string;
    password: string;
    username: string;
    branch: string;
}

export class MultiClusterRepositoryUpdateRequest extends BaseRequest {
    gitTimeout: number;
    syncInterval: number;
    syncEnable: boolean;
}

export class TreeNode {
    name: string;
    path: string;
    dir: boolean;
    children: TreeNode[] = [];
    content: string;
    originContent: string;
    active: boolean;
    changed: boolean;
    delete = false;
}

export class FileContent {
    content: string;
    path: string;
    readOnly = false;
}

export class ReadFileRequest {
    fileName: string;
}

export class RelatedCluster {
    clusterName: string;
    status: string;
    message: string;
    createdAt: string;
}

export class RelateClusterRequest {
    clusterNames: string[] = [];
    delete = false;
}

export class MultiClusterSyncLog extends BaseModel {
    id: string;
    gitCommitId: string;
    status: string;
}

export class ResourceLog extends BaseModel {
    resourceName: string;
    sourceFile: string;
    status: string;
    message: string;

}

export class MultiClusterSyncClusterLog extends BaseModel {
    clusterName: string;
    multiClusterSyncClusterResourceLogs: ResourceLog[] = [];
}


export class MultiClusterSyncLogDetail extends BaseModel {
    multiClusterSyncClusterLogs: MultiClusterSyncClusterLog[] = [];
}
